src/controllers/aer_controller.hpp

```diff
@@ -1962,6 +1695,170 @@ void Controller::measure_sampler(
   }
 }
 
+//-------------------------------------------------------------------------
+// Validation
+//-------------------------------------------------------------------------
+
+std::vector<Controller::Method>
+Controller::simulation_methods(std::vector<Circuit> &circuits,
+                               Noise::NoiseModel &noise_model) const {
+  // Does noise model contain kraus noise
+  bool kraus_noise = (noise_model.opset().contains(Operations::OpType::kraus) ||
+                      noise_model.opset().contains(Operations::OpType::superop));
+
+  if (default_method_ == Method::automatic) {
+    // Determine simulation methods for each circuit and noise model
+    std::vector<Method> sim_methods;
+    bool superop_enabled = false;
+    bool kraus_enabled = false;
+    for (const auto& circ: circuits) {
+      auto method = automatic_simulation_method(circ, noise_model);
+      sim_methods.push_back(method);
+      if (!superop_enabled && (method == Method::density_matrix || method == Method::superop)) {
+        noise_model.enable_superop_method(max_parallel_threads_);
+        superop_enabled = true;
+      } else if (kraus_noise && !kraus_enabled &&
+                 (method == Method::statevector || method == Method::matrix_product_state)) {
+        noise_model.enable_kraus_method(max_parallel_threads_);
+        kraus_enabled = true;
+      }
+    }
+    return sim_methods;
+  }
+
+  // Use non-automatic default method for all circuits
+  std::vector<Method> sim_methods(circuits.size(), default_method_);
+  if (default_method_ == Method::density_matrix || default_method_ == Method::superop) {
```

Comment:
I guess that `validate_memory_requirements`, `validate_state`, and `check_measure_sampling_opt` are necessary for each circuit as `automatic_simulation_method` does.